packages/ods-react/src/components/tree-view/src/components/tree-view-node/TreeViewNode.tsx

```diff
@@ -0,0 +1,208 @@
+  customData?: CustomData;
+  isBranch: boolean;
+  isExpanded: boolean;
+  item: TreeViewItem<CustomData>;
+  name: string;
+};
+
+interface TreeViewNodeProp<CustomData = Record<string, never>> extends Omit<ComponentPropsWithRef<'div'>, 'children'> {
+  children?: ReactNode | ((arg: TreeViewCustomRendererArg<CustomData>) => JSX.Element);
+  indexPath?: number[];
+  item: TreeViewItem<CustomData>;
+}
+
+const TreeViewNode: FC<TreeViewNodeProp> = forwardRef(({
+  children,
+  className,
+  indexPath,
+  item,
+  ...props
+}, ref): JSX.Element => {
+  const { multiple, disabled: globalDisabled } = useTreeView();
+  const { indexPath: contextIndexPath } = useTreeViewNodes();
+  const effectiveIndexPath = indexPath ?? contextIndexPath;
+
+  const isBranch = !!item.children?.length;
+  const isDisabled = !!(globalDisabled || item.disabled);
+  const processedItem = {
+    ...item,
+    disabled: isDisabled,
+  };
+  const checkboxRef = useRef<HTMLLabelElement | null>(null);
+  function handleKeyDown(e: KeyboardEvent<HTMLElement>): void {
+    if (!multiple || isDisabled) {
+      return;
+    }
+    if (e.key === ' ' || e.key === 'Enter') {
+      e.preventDefault();
+      e.stopPropagation();
+      checkboxRef.current?.click();
+    }
+  }
+  function renderCheckboxIndicator(isDisabled: boolean): JSX.Element {
+    return (
+      <VendorTreeView.NodeCheckboxIndicator
+        fallback={
+          <Checkbox checked={ false } disabled={ isDisabled }>
+            <CheckboxControl />
+          </Checkbox>
+        }
+        indeterminate={
+          <Checkbox checked={ 'indeterminate' } disabled={ isDisabled }>
+            <CheckboxControl />
+          </Checkbox>
+        }>
+        <Checkbox checked disabled={ isDisabled }>
+          <CheckboxControl />
+        </Checkbox>
+      </VendorTreeView.NodeCheckboxIndicator>
+    );
+  }
+  function renderLabel(): JSX.Element {
+    if (typeof children === 'function') {
+      const renderer = children as (arg: TreeViewCustomRendererArg<unknown>) => JSX.Element;
+      const content = renderer({
+        customData: item.customRendererData,
+        isBranch: !!item.children?.length,
+        isExpanded: false,
+        item,
+        name: item.name,
+      });
+
+      if (isBranch) {
+        return <VendorTreeView.BranchText>{ content }</VendorTreeView.BranchText>;
+      }
+
+      return multiple ? (
+        <VendorTreeView.ItemText>{ content }</VendorTreeView.ItemText>
+      ) : (
+        <VendorTreeView.ItemText asChild>
+          <span>{ content }</span>
+        </VendorTreeView.ItemText>
+      );
+    }
+
+    return isBranch ? (
+      <VendorTreeView.BranchText>{ item.name }</VendorTreeView.BranchText>
+    ) : (
+      <VendorTreeView.ItemText>{ item.name }</VendorTreeView.ItemText>
+    );
+  }
+
+  const baseClassName = isBranch
+    ? classNames(style['tree-view-node'], className)
+    : classNames(style['tree-view-node'], style['tree-view-node__item'], className);
```

Comment:
use classNames conditionnal syntax directly in the JSX to avoid replication of classes